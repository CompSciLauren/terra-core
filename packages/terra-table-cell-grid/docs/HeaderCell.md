# Header Cell

The Terra TableCellGrid Header Cell is a sub component of the table header. The header cell dictates the minimum width of the column, but it can also dictate the maximum width of the column if a fixed or max-width value is set it.

While sort indicators are presented by the header cell, the actual sorting logic needs to be handled by the consumer by implementing the `onSelect` and `metaData` props.

## Usage

```jsx
import React from 'react';
import { HeaderCell } from 'terra-table-cell-grid';

<HeaderCell sort="asc" minWidth="small">
  Column 0
</HeaderCell>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
