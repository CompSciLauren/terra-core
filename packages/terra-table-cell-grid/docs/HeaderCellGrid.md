# Header Cell Grid

The HeaderCellGrid is a sub component that groups header cells passed through children. Each cell grid handles it's own selection state through props. Can be set to synchronize space with CellGrids using check mark or chevron style selection.

## Usage

```jsx
import React from 'react';
import { HeaderCellGrid, CellGrid } from 'terra-table-cell-grid';

  <HeaderCellGrid
    key="unique-react-key"
  >
    <CellGrid>
      Cell Content
    </CellGrid>
    <CellGrid>
      Cell Content
    </CellGrid>
  </HeaderCellGrid>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
