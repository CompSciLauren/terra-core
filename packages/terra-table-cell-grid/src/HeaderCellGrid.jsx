import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import HeaderCheckMarkCell from './HeaderCheckMarkCell';
import HeaderChevronCell from './HeaderChevronCell';

import styles from './TableCellGrid.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The children passed to the component
   */
  children: PropTypes.node,
  /**
   * Function callback for the ref of the tr.
   */
  refCallback: PropTypes.func,
  /**
   * Style of selection for the row.
   * One of `default`, `checkmark`, `chevron`.
   */
  selectionStyle: PropTypes.oneOf(['default', 'checkmark', 'chevron']),
};

const defaultProps = {
  children: [],
  selectionStyle: 'default',
};

const HeaderCellGrid = ({
  children,
  refCallback,
  selectionStyle,
  ...customProps
}) => {
  const rowClassNames = cx([
    'header',
    customProps.className,
  ]);

  let checkmark;
  let chevron;
  if (selectionStyle === 'checkmark') {
    checkmark = (
      <HeaderCheckMarkCell isPadded />
    );
  } else if (selectionStyle === 'chevron') {
    chevron = (
      <HeaderChevronCell isPadded />
    );
  }

  return (
    <div className={rowClassNames} role="rowgroup">
      <div className={cx(['header-content'])} role="row">
        {checkmark}
        {children}
        {chevron}
      </div>
    </div>
  );
};

HeaderCellGrid.propTypes = propTypes;
HeaderCellGrid.defaultProps = defaultProps;

export default HeaderCellGrid;
