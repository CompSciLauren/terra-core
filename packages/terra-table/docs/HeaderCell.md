# Header Cell

The Terra Table Header Cell is a sub component of the table header.

While sort indicators are presented by the header cell, the actual sorting logic needs to be handled by the consumer by implementing the `onSelect` and `metaData` props.

## Usage

```jsx
import React from 'react';
import { HeaderCell } from 'terra-table';

<HeaderCell sort="asc">
  Column 0
</HeaderCell>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
