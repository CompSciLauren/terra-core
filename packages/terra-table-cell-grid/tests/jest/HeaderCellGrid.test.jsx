import React from 'react';
import HeaderCellGrid from '../../src/HeaderCellGrid';
import HeaderCell from '../../src/HeaderCell';

// Snapshot test
it('should render a default header cell grid', () => {
  const shallowComponent = shallow(<HeaderCellGrid />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a cell grid with a cell', () => {
  const shallowComponent = shallow(<HeaderCellGrid><HeaderCell /></HeaderCellGrid>);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a checkmark selectionStyle cell grid', () => {
  const shallowComponent = shallow(<HeaderCellGrid selectionStyle="checkmark" />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a chevron selectionStyle cell grid', () => {
  const shallowComponent = shallow(<HeaderCellGrid selectionStyle="chevron" />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a cell grid with refCallback', () => {
  const shallowComponent = shallow(
    <HeaderCellGrid refCallback={jest.fn()} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});
