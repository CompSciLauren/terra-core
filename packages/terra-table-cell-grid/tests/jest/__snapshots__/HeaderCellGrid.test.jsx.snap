// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a cell grid with a cell 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  >
    <HeaderCell
      isSelectable={false}
      sort="none"
    />
  </div>
</div>
`;

exports[`should render a cell grid with refCallback 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  />
</div>
`;

exports[`should render a checkmark selectionStyle cell grid 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  >
    <HeaderCheckMarkCell
      isIntermediate={false}
      isPadded={true}
      isSelectable={false}
      isSelected={false}
    />
  </div>
</div>
`;

exports[`should render a chevron selectionStyle cell grid 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  >
    <HeaderChevronCell
      isPadded={true}
    />
  </div>
</div>
`;

exports[`should render a default header cell grid 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  />
</div>
`;

exports[`should render a selectable cell grid 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  />
</div>
`;
